logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/PersistOtherUserClientsUseCase.kt

```diff
@@ -0,0 +1,33 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logger.KaliumLogger.Companion.ApplicationFlow.CLIENTS
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.client.DeviceType
+import com.wire.kalium.logic.data.client.remote.ClientRemoteRepository
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.network.api.UserId as NetworkUserID
+
+interface PersistOtherUserClientsUseCase {
+    suspend operator fun invoke(userId: UserId)
+}
+
+internal class PersistOtherUserClientsUseCaseImpl(
+    private val clientRemoteRepository: ClientRemoteRepository,
+    private val clientRepository: ClientRepository
+) : PersistOtherUserClientsUseCase {
+    override suspend operator fun invoke(userId: UserId): Unit =
+        clientRemoteRepository.otherUserClients(NetworkUserID(userId.value, userId.domain)).fold({
```

Comment:
I'm not pretty sure about having this `NetworkUserID` at this level, usually we delegate to repositories to do the mapping work, and we usually pass domain classes to repos, not lower level dependencies.

In other words, we should pass to the repo, the `UserId` and delegate this mapping in there :D